pkg/operator/ceph/disruption/clusterdisruption/osd.go

```diff
@@ -261,18 +257,15 @@ func (r *ReconcileClusterDisruption) reconcilePDBsForOSDs(
 		delete(pdbStateMap.Data, drainingFailureDomainDurationKey)
 		delete(pdbStateMap.Data, pgHealthCheckDurationKey)
 	} else {
-		logger.Infof("all osds are up. last drained failure domain: %q.  pg health: %q. waiting for PGs to be active+clean or PG health check to timeout",
+		logger.Infof("last drained failure domain: %q. pg health: %q. waiting for PGs to be active+clean or PG health check to timeout",
 			pdbStateMap.Data[drainingFailureDomainKey], pgHealthMsg)
 	}
 
-	err = r.updateNoout(clusterInfo, pdbStateMap, allFailureDomains)
-	if err != nil {
-		logger.Errorf("failed to update maintenance noout in cluster %q. %v", request, err)
-	}
-
-	err = r.client.Update(context.TODO(), pdbStateMap)
-	if err != nil {
-		return errors.Wrapf(err, "failed to update configMap %q in cluster %q", pdbStateMapName, request)
+	if pdbStateMap.Data[setNoOut] == "true" {
+		err = r.updateNoout(clusterInfo, pdbStateMap, allFailureDomains)
+		if err != nil {
+			logger.Errorf("failed to update maintenance noout in cluster %q. %v", request, err)
+		}
 	}
 
 	// handle inactive drains by deleting all blocking pdb and restoring the main pdb
```

Comment:
What are inactive drains? This just means the default PDBs are restored, right?

Reply:
Yes, I'll change the verbiage here.